modules/backend/formwidgets/repeater/partials/_repeater.htm

```diff
@@ -14,9 +15,10 @@
 
     <div class="field-repeater-add-item loading-indicator-container indicator-center">
         <a
-            href="javascript:;"
-            data-request="<?= $this->getEventHandler('onAddItem') ?>"
-            data-load-indicator>
+                href="javascript:;"
+                data-request="<?= $this->getEventHandler('onAddItem') ?>"
+                data-request-complete="($('.field-repeater').attr('data-max-items') > $('.field-repeater-item').length || $('.field-repeater').attr('data-max-items') == 0) ? $('.field-repeater-add-item').show() : $('.field-repeater-add-item').hide()"
```

Comment:
I feel like this could be handled cleaner, within the `repeater.js` logic, instead of on the element itself. I'd like you to try hooking into the AJAX framework events http://octobercms.com/docs/ajax/javascript-api#global-events